src/qonnx/transformation/quantize_graph.py

```diff
@@ -0,0 +1,261 @@
+from qonnx.transformation.general import SortGraph
+from qonnx.transformation.infer_shapes import InferShapes
+from qonnx.util.basic import qonnx_make_model
+from qonnx.util.cleanup import cleanup_model
+
+
+class graph_util:
+    def get_node_id(self, model):
+        node_index = {}
+        node_ind = 0
+        for node in model.graph.node:
+            node_index[node.name] = node_ind
+            node_ind += 1
+        return node_index
+
+    def node_from_name(self, model, node_name):
+        for node in model.graph.node:
+            if node.name == node_name:
+                return node
+
+    def identify_nodes(self, model, node_type):
+        node_list = []
+        for node in model.graph.node:
+            if node.op_type == node_type:
+                node_list.append(node)
+        return node_list
+
+    def create_node(
+        self,
+        model,
+        quantnode_input,
+        quantnode_output_shape,
+        node_count,
+        tensor_count,
+        scale_value,
+        zeropoint_value,
+        bitwidth_value,
+        narrow,
+        signed,
+        rounding_mode,
+    ):
+        quantnode_output_dtype = DataType["UINT8"]
+        quant_tensor = onnx.helper.make_tensor_value_info(
+            model.make_new_valueinfo_name(), TensorProto.FLOAT, quantnode_output_shape
+        )
+        model.graph.value_info.append(quant_tensor)
+        model.set_tensor_datatype(quant_tensor.name, quantnode_output_dtype)
+
+        stationary_input_dtype = DataType["FLOAT32"]
+        scale_tensor = np.array(scale_value).astype(np.float32)
+        s_value = onnx.helper.make_tensor_value_info(
+            model.make_new_valueinfo_name(), TensorProto.FLOAT, quantnode_output_shape
+        )
+        model.graph.value_info.append(s_value)
+        model.set_tensor_datatype(s_value.name, stationary_input_dtype)
+        model.set_initializer(s_value.name, scale_tensor)
+
+        zeropt_tensor = np.array(zeropoint_value).astype(np.float32)
+        z_value = onnx.helper.make_tensor_value_info(
+            model.make_new_valueinfo_name(), TensorProto.FLOAT, quantnode_output_shape
+        )
+        model.graph.value_info.append(z_value)
+        model.set_tensor_datatype(z_value.name, stationary_input_dtype)
+        model.set_initializer(z_value.name, zeropt_tensor)
+
+        bitwidth_tensor = np.array(bitwidth_value).astype(np.float32)
+        b_value = onnx.helper.make_tensor_value_info(model.make_new_valueinfo_name(), TensorProto.FLOAT, [1])
+        model.graph.value_info.append(b_value)
+        model.set_tensor_datatype(b_value.name, stationary_input_dtype)
+        model.set_initializer(b_value.name, bitwidth_tensor)
+
+        quant_node = onnx.helper.make_node(
+            "Quant",
+            inputs=[quantnode_input, s_value.name, z_value.name, b_value.name],
+            outputs=[quant_tensor.name],
+            name="Quant_node_" + str(node_count) + str(tensor_count),
```

Comment:
would suggest a slightly more descriptive naming scheme for newly created Quant nodes, e.g.

`Quant_<name of target node>_<inp or out>_<index>`